menas/src/main/scala/za/co/absa/enceladus/rest/controllers/MappingTableController.scala

```diff
@@ -34,17 +35,25 @@ class MappingTableController @Autowired() (mappingTableService: MappingTableServ
   import scala.concurrent.ExecutionContext.Implicits.global
 
   @PostMapping(path = Array("/updateDefaults"))
-  def updateDefaults(@AuthenticationPrincipal user: UserDetails, @RequestBody upd: MenasObject[Array[DefaultValue]]): CompletableFuture[MappingTable] = {
+  def updateDefaults(@AuthenticationPrincipal user: UserDetails,
+                     @RequestBody upd: MenasObject[Array[DefaultValue]]): CompletableFuture[ResponseEntity[MappingTable]] = {
     mappingTableService.update(user.getUsername, upd.id.name) { mt =>
       mt.setDefaultMappingValue(upd.value.toList)
+    }.map {
+      case Some(entity) => ResponseEntity.ok().body(entity)
```

Comment:
```
    ResponseEntity.ok().body(entity)
    ResponseEntity.ok(entity)
    new ResponseEntity(entity, HttpStatus.OK)
```
If you have no reason to use all of them then it would be nice to have only one. It causes lot of confusion.
And because of responses like: ` ResponseEntity.notFound().build[MappingTable]()` 
It would be better define your own "response codes" that will wrap builders. 
For example instead of this: ` ResponseEntity.notFound().build[MappingTable]()` can be NotFound[MappingTable]

Reply:
Really appreciate the feedback! I used to do it this way in Play, just got lazy here or curious to see how it feels doing it like this. The way we build `201 CREADTED` sticks out the most, since it accepts a location header, which we can't really supply without bending over backwards. Ideally we would provide the location header, but currently we return the created entity and that is enough for now.

Thanks for keeping me accountable!